charts/flyte-binary/templates/auth-client-secret.yaml

```diff
@@ -0,0 +1,19 @@
+{{- if .Values.configuration.auth.enabled }}
+apiVersion: v1
+kind: Secret
+metadata:
+  name: {{ include "flyte-binary.configuration.auth.clientSecretName" . }}
+  namespace: {{ .Release.Namespace | quote }}
+  labels: {{- include "flyte-binary.labels" . | nindent 4 }}
+    {{- if .Values.commonLabels }}
+    {{- include "flyte-binary.renderTemplate" ( dict "value" .Values.commonLabels "context" $ ) | nindent 4 }}
```

Comment:
remind me to ask you how this works tomorrow.

Reply:
This is basically a helper for rendering template strings: https://github.com/flyteorg/flyte/blob/0ba5bc05c02f6d169543c2495f3c4ac36b5cf9f3/charts/flyte-binary/templates/_helpers.tpl#L65.

`( dict "value" .Values.commonLabels "context" $ )` generates a map with 2 keys: `value` and `context`. This is then passed into the template. The output is `.value` rendered with attributes from `.context`, with handling for simple strings and more complex structures. I saw this pattern in one of the bitnami charts to standardize how templates are rendered.

Reply:
In retrospect, this is needlessly verbose. I'll refactor this out a follow-up PR.

Reply:
https://github.com/flyteorg/flyte/pull/3156